packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -521,6 +538,37 @@ impl ScriptTransactionBuilder {
         Ok(tx)
     }
 
+    fn resolve_fuel_tx_for_state_reading(self, dry_runner: impl DryRunner) -> Result<Script> {
+        let num_resolved_witnesses = self.num_witnesses()?;
+        let mut script_dry_runner = self.script_dry_runner(num_resolved_witnesses, &dry_runner);
+
+        let mut tx = FuelTransaction::script(
+            0, // default value - will be overwritten
+            self.script.clone(),
+            self.script_data.clone(),
+            self.generate_fuel_policies()?,
+            resolve_fuel_inputs(
+                self.inputs.clone(),
+                num_resolved_witnesses,
+                &self.unresolved_witness_indexes,
+            )?,
+            self.outputs.clone(),
+            vec![],
+        );
+
+        let should_saturate_variable_outputs =
```

Comment:
Why do we need to care about variable outputs if it's read only operation?

Reply:
Theoretically if it only were a simple read only we wouldn't. 

Was thinking more about some more complex logic being runnable in this way. I.e. a function mints but also returns something from storage after doing so. 

If we don't propagate the behavior of the variable output policy we're bound to get complaints about not being able to "read only" the tx. Also we might get complaints that we don't respect wishes, 2 lines above the policy was set to exactly 100, but we don't take that into account when simulating in this way.

If the policy isn't explicitly set it defaults to `Exactly(0)` so we won't add any variable outputs in the default case. 

If the user sets a policy it's going to be reflected in this simulation mode the same way it would have been reflected in the other -- because without it you might not get to the end of the contract method.